StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1132CodeFixProvider.cs

```diff
@@ -0,0 +1,127 @@
+    /// </remarks>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(SA1132CodeFixProvider))]
+    [Shared]
+    internal class SA1132CodeFixProvider : CodeFixProvider
+    {
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(SA1132DoNotCombineFields.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return CustomFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        ReadabilityResources.SA1132CodeFix,
+                        cancellationToken => GetTransformedDocumentAsync(context.Document, diagnostic, cancellationToken),
+                        equivalenceKey: nameof(SA1132CodeFixProvider)),
+                    diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var baseFieldDeclaration = (BaseFieldDeclarationSyntax)syntaxRoot.FindNode(diagnostic.Location.SourceSpan);
+            List<BaseFieldDeclarationSyntax> newFieldDeclarations = SplitDeclaration(baseFieldDeclaration);
+
+            if (newFieldDeclarations != null)
+            {
+                var editor = new SyntaxEditor(syntaxRoot, document.Project.Solution.Workspace);
+                editor.InsertAfter(baseFieldDeclaration, newFieldDeclarations);
+                editor.RemoveNode(baseFieldDeclaration);
+                return document.WithSyntaxRoot(editor.GetChangedRoot());
+            }
+
+            return document;
+        }
+
+        private static List<BaseFieldDeclarationSyntax> SplitDeclaration(BaseFieldDeclarationSyntax baseFieldDeclaration)
+        {
+            var fieldDeclaration = baseFieldDeclaration as FieldDeclarationSyntax;
+            if (fieldDeclaration != null)
+            {
+                VariableDeclarationSyntax declaration = fieldDeclaration.Declaration;
+                SeparatedSyntaxList<VariableDeclaratorSyntax> variables = declaration.Variables;
+                VariableDeclaratorSyntax first = variables.First();
+                var newFieldDeclarations = new List<BaseFieldDeclarationSyntax>(variables.Count);
+
+                foreach (VariableDeclaratorSyntax variable in variables)
+                {
+                    var variableDeclarator = SyntaxFactory.SeparatedList(new[] { variable });
```

Comment:
:bulb: The downside of this syntax is it forces the code to use this overload even if a future release includes an overload of `SeparatedList` that takes exactly one argument.